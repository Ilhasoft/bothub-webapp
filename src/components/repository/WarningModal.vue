<template>
  <bh-modal :open.sync="openValue">
    <div class="bh-grid">
      <div class="bh-grid__item">
        <h2>Warnings</h2>
        <div
          v-for="(warnings, lang) in languagesWarnings"
          :key="lang">
          <div><strong>{{ lang | languageVerbose }}</strong></div>
          <div
            v-for="(warning, index) in warnings"
            :key="index">{{ warning }}</div>
        </div>
      </div>
    </div>
  </bh-modal>
</template>

<script>
export default {
  name: 'WarningModal',
  props: {
    open: {
      type: Boolean,
      default: false,
    },
    languagesWarnings: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      openValue: this.open,
    };
  },
  watch: {
    open(value) {
      this.openValue = value;
    },
    openValue(value) {
      this.$emit('update:open', value);
    },
  },
};
</script>
