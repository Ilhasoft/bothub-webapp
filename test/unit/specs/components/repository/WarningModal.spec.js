import { shallowMount, createLocalVue } from '@vue/test-utils';
import BH from 'bh';
import WarningModal from '@/components/repository/WarningModal';
import applyFilters from '@/utils/filters';


const localVue = createLocalVue();
localVue.use(BH);
applyFilters(localVue);

describe('WarningModal.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(WarningModal, {
      localVue,
      mocks: {
        $t: () => 'some specific text',
      },
      propsData: {
        open: true,
        languagesWarnings: {},
        training: false,
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('add warning', () => {
    beforeEach(() => {
      wrapper.setProps({
        languagesWarnings: {
          en: [
            'warning 1',
          ],
        },
      });
    });

    test('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('close modal via prop', () => {
    beforeEach(() => {
      wrapper.setProps({ open: false });
    });

    test('event update:open was emitted', () => {
      expect(wrapper.emitted('update:open')).toBeDefined();
    });
  });
});
