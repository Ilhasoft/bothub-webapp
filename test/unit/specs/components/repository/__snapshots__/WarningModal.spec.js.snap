// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningModal.vue add warning renders correctly 1`] = `
<bh-modal-stub open="true" title="" backgroundcloses="true">
  <div class="bh-grid">
    <div class="bh-grid__item">
      <h2>Warnings</h2>
      <div>
        <div><strong>English</strong></div>
        <div>warning 1</div>
      </div>
    </div>
  </div>
</bh-modal-stub>
`;

exports[`WarningModal.vue renders correctly 1`] = `
<bh-modal-stub open="true" title="" backgroundcloses="true">
  <div class="bh-grid">
    <div class="bh-grid__item">
      <h2>Warnings</h2>
    </div>
  </div>
</bh-modal-stub>
`;
